src/gsi/m_extOzone.F90

```diff
@@ -1421,13 +1439,286 @@ subroutine ozlev_bufrread_(dfile,dtype,dsis, ozout,nmrecs,ndata,nodata, &
+        do ilev = 1, levs
+           if (ozone(ilev, iprof) .lt. -900.0) cycle ! undefined
+        end do
+!!$           if (iuse_oz(ipos(ilev)) < 0) then
+!!$              usage = 10000._r_kind
+!!$           else
+!!$              usage = zero
+!!$           endif
+        nmrecs=nmrecs+levs+1
+
+        ! convert observation location to radians
+        slons0=slonsa(iprof)
+        slats0=slatsa(iprof)
+        if(abs(slats0)>90._r_kind .or. abs(slons0)>r360) cycle
+        if(slons0< zero) slons0=slons0+r360
+        if(slons0==r360) slons0=zero
+        dlat_earth_deg = slats0
+        dlon_earth_deg = slons0
+        dlat_earth = slats0 * deg2rad
+        dlon_earth = slons0 * deg2rad
+        
+        if(regional)then
+           call tll2xy(dlon_earth,dlat_earth,dlon,dlat,outside)
+           if(outside) cycle    
+        else
+           dlat = dlat_earth
+           dlon = dlon_earth
+           call grdcrd1(dlat,rlats,nlat,1)
+           call grdcrd1(dlon,rlons,nlon,1)
+        endif
+        
+        idate5(1) = iya(iprof) !year
+        idate5(2) = ima(iprof) !month
+        idate5(3) = idda(iprof) !day
+        idate5(4) = ihha(iprof) !hour
+        idate5(5) = imina(iprof) !minute
+        call w3fs21(idate5,nmind)
+        t4dv=real((nmind-iwinbgn),r_kind)*r60inv
+        if (l4dvar.or.l4densvar) then
+           if (t4dv<zero .OR. t4dv>winlen) then
+              write(6,*)'read_ozone: ', dtype,' obs time idate5=',idate5,', t4dv=',&
+                   t4dv,' is outside time window, sstime=',sstime*r60inv
+              cycle
+           end if
+        else
+           sstime=real(nmind,r_kind)
+           tdiff=(sstime-gstime)*r60inv
+           if(abs(tdiff) > twind)then
+              write(6,*)'read_ozone: ',dtype,' obs time idate5=',idate5,', tdiff=',&
+                   tdiff,' is outside time window=',twind
+              cycle
+           end if
+        end if
+        
+        !! Compute total ozone
+        totoz=zero
+        do k=1,levs
+           poz(k) = ozone(k,iprof)
+           totoz=totoz+ozone(k,iprof)
+        end do
+        poz(levs+1) = totoz
+        
+        !Check ozone layer values.  If any layer value is bad, toss entire profile
+        do k=1,levs
+           if (poz(k)>badoz) cycle read_loop1
+        end do
+        
+        !       Write ozone record to output file
+        ndata=min(ndata+1,maxobs)
+        if(ndata<=maxobs) then
+           nodata=nodata+levs+1
+           ozout(1,ndata)=rsat
+           ozout(2,ndata)=t4dv
+           ozout(3,ndata)=dlon               ! grid relative longitude
+           ozout(4,ndata)=dlat               ! grid relative latitude
+           ozout(5,ndata)=dlon_earth_deg     ! earth relative longitude (degrees)
+           ozout(6,ndata)=dlat_earth_deg     ! earth relative latitude (degrees)
+           ozout(7,ndata)=0.                 ! total ozone error flag
+           ozout(8,ndata)=0.                 ! profile ozone error flag
```

Comment:
Add `_r_kind` suffix to `0.0` or use `zero` from `constants` module. 
 (GSI coding standard).